component/messageHandler.go

```diff
@@ -0,0 +1,96 @@
+package component
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	p_buff "th2-grpc/th2_grpc_common"
+
+	"github.com/rs/zerolog/log"
+	rabbitmq "github.com/th2-net/th2-common-go/schema/modules/mqModule"
+	"github.com/th2-net/th2-common-go/schema/queue/MQcommon"
+)
+
+type MessageTypeListener struct {
+	MessageType    string
+	Function       func(args ...interface{})
+	RootEventID    *p_buff.EventID
+	Module         *rabbitmq.RabbitMQModule
+	AmountReceived *int
+	NBatches       int
+	Stats          map[string]int
+}
+
+func NewListener(RootEventID *p_buff.EventID, module *rabbitmq.RabbitMQModule, BoxConf *BoxConfiguration, Function func(args ...interface{})) *MessageTypeListener {
+	amountReceived := 0
+	return &MessageTypeListener{
+		MessageType:    BoxConf.MessageType,
+		Function:       Function,
+		RootEventID:    RootEventID,
+		Module:         module,
+		AmountReceived: &amountReceived,
+		NBatches:       4,
+		Stats:          make(map[string]int),
```

Comment:
Please use several counter instead of map in case where we know about all key values